test/integration/smoke/test_snapshots.py

```diff
@@ -245,23 +245,30 @@ def test_02_list_snapshots_with_removed_data_store(self):
             PASS,
             "Invalid response returned for list volumes")
         vol_uuid = vol_res[0].id
-        
-        # Create new Primary Storage
         clusters = list_clusters(
             self.apiclient,
             zoneid=self.zone.id
         )
         assert isinstance(clusters,list) and len(clusters)>0
 
+        self.cleanup.append(self.virtual_machine_with_disk)
```

Comment:
this is not created here. I applaud the move , but should't it be registered for cleanup right after creation to prevent it being left if an exception occurs in the meanwhile?
also as it is created at class level it should go in `_cleanup` and not in `cleanup`.